### PR TITLE
handles multiple directories from `get_imaging_root_data_dir()` & code-cleanup

### PR DESCRIPTION
**Pull request by ttngu207**

### CODE REVIEW

File: element_calcium_imaging/__init__.py [hunk 1]
```diff
@@ -1,5 +1,55 @@
...
+        raise FileNotFoundError(f'{full_path} does not exist!')
+
+    # turn to list if only a single root directory is provided
+    if isinstance(root_directories, (str, pathlib.Path)):
+        root_directories = [root_directories]
+
+    try:
+        return next(pathlib.Path(root_dir) for root_dir in root_directories
+                    if pathlib.Path(root_dir) in set(full_path.parents))
+
```
Comment (dimitri-yatsenko) [inline element_calcium_imaging/__init__.py:52]: remove this blank line. The next block is not a new logical block

File: Background.md [hunk 2]
```diff
@@ -22,9 +22,8 @@ In all there are perhaps on the order of 3000 two-photon setups globally but the
 + ThorImageLS
 + ScanBox
 
-Vidrio’s ScanImage is the data acquisition software for two types of home-built scanning two-photon systems: 
+Vidrio’s [ScanImage](https://docs.scanimage.org/) is the data acquisition software for two types of home-built scanning two-photon systems:
```
Comment (shenshan) [inline Background.md:25]:
```suggestion
Vidrio’s [ScanImage](https://docs.scanimage.org/) is the data acquisition software for two types of home-built scanning two-photon systems, either based on Thorlabs or Sutter hardware.
```

File: Background.md [hunk 3]
```diff
@@ -22,9 +22,8 @@ In all there are perhaps on the order of 3000 two-photon setups globally but the
 + ThorImageLS
 + ScanBox
 
-Vidrio’s ScanImage is the data acquisition software for two types of home-built scanning two-photon systems: 
+Vidrio’s [ScanImage](https://docs.scanimage.org/) is the data acquisition software for two types of home-built scanning two-photon systems: 
 based on Thorlabs and Sutter hardware. ScanImage has a free version and a licensed version.
```
Comment (shenshan) [inline Background.md:26]:
```suggestion
```

File: element_calcium_imaging/__init__.py [hunk 4]
```diff
@@ -1,5 +1,54 @@
...
+def find_full_path(root_directories, relative_path):
+    """
+    Given a relative path, search and return the full-path
+     from provided potential root directories (in the given order)
+        :param root_directories: potential root directories
+        :param relative_path: the relative path to find the valid root directory
+        :return: root_directory (pathlib.Path object)
+    """
+    relative_path = pathlib.Path(relative_path)
+
```
Comment (shenshan) [inline element_calcium_imaging/__init__.py:16]:
```suggestion
# If the relative_path is an absolute path
```

File: element_calcium_imaging/imaging.py [hunk 5]
```diff
@@ -128,7 +112,7 @@ class MaskType(dj.Lookup):
 
 @schema
 class ProcessingTask(dj.Manual):
-    definition = """
+    definition = """  #
```
Comment (shenshan) [inline element_calcium_imaging/imaging.py:115]:
```suggestion
    definition = """  #  Manual table for defining a processing task to be run.
```

**Review by kabilar — APPROVED** [review]

**Pull request merged**